### DISCUSSION TITLE
Store Client State in database

### DISCUSSION THREAD

**Discussion by tclasen** [idom-team/idom#756] [Question]
For deployment scenarios where the server is short-lived, storing client state in the server memory would pose problems. Would it be possible to instead store this state in a database and use reactivity magic to connect the two?

**Comment by Archmonger** [comment]
I have a [proposal for this](https://github.com/idom-team/django-idom/issues/39) sitting within Django IDOM's repository. It is planned but my OSS dev time has been limited lately.

Once it's developed for Django IDOM it will likely get propogated to the other web frameworks.

**Reply by tclasen** [reply]
To decouple my multiple questions. What happens when I have 3 replicas of my server running and the one holding my client state dies. Will the client try to connect to another server? Will it have to rebuild the Dom from scratch? Will it be a jarring experience for the user?

**Reply by rmorshea** [reply]
The client does attempt to reconnect, but we don't have a visual indicator of this presently. However, upon reconnecting, the client does unmount and then remount a new view resulting in a brief flash as that new view renders . This isn't strictly necessary though, and we could certainly improve this re-connection behavior so the client does not perform a full unmount of the old view.

**Reply by rmorshea** [reply]
I've created two new issues to capture these improvements: #760 #761
